### PR TITLE
Economy API Documentation

### PR DESCRIPTION
**Pull request by Flibio**
Documentation for the Economy API. Any input is appreciated. 

Features of this PR:

- [x] Documents Economy API basics
- [x] Gets you on the right path to implement the Economy API
- [x] Describes and provides examples for using the Economy API

### CODE REVIEW

File: source/plugin/economy/basics.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Basics
+======
+
+The Sponge Economy API has a few basic components that developers should be familiar with: 
+accounts, transactions, Currency, and the EconomyService itself. Each part will be discussed below.
+
+EconomyService
+--------------
+
+The EconomyService is the basis of the Economy API. It stores and economy's currencies, and provides methods
```
Comment (Aaron1011) [inline source/plugin/economy/basics.rst]: You should mention that there's no default implementation.

File: source/plugin/economy/basics.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+======
+
+The Sponge Economy API has a few basic components that developers should be familiar with: 
+accounts, transactions, Currency, and the EconomyService itself. Each part will be discussed below.
+
+EconomyService
+--------------
+
+The EconomyService is the basis of the Economy API. It stores and economy's currencies, and provides methods 
+for account management. Note that there is no default implementation for the EconomyService. The EconomyService
```
Comment (Inscrutable) [inline source/plugin/economy/basics.rst]: "It stores all currencies" looks better. Either that or change and -> an

File: source/plugin/economy/using.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	import org.spongepowered.api.event.service.ChangeServiceProviderEvent;
+	import org.spongepowered.api.service.economy.EconomyService;
+
+	private EconomyService economyService;
+	
+	@Listener
+	public void onChangeServiceProvider(ChangeServiceProviderEvent event) {
+		if(event.getService().equals(EconomyService.class)) {
+			Object rawProvider = event.getNewProviderRegistration().getProvider();
+			if(rawProvider instanceof EconomyService) {
```
Comment (phroa) [inline source/plugin/economy/using.rst]: Spaces after both these `if`s

File: source/plugin/economy/using.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+.. code-block:: java
+	
+	import org.spongepowered.api.entity.living.player.Player;
+	import org.spongepowered.api.service.economy.EconomyService;
+	import org.spongepowered.api.service.economy.account.UniqueAccount;
+	import java.math.BigDecimal;
+	import java.util.Optional;
+
+	Optional<UniqueAccount> uOpt = economyService.getAccount(player.getUniqueId());
+	if(uOpt.isPresent()) {
```
Comment (phroa) [inline source/plugin/economy/using.rst]: space after `if`

**Comment by saladoc** [comment]
I don't think we actually need examples for implementing the API. Everyone attempting to do that should have a pretty solid idea about it if he wants to succeed. I think we're fine if the implementation docs outline what each method is supposed to do, where known caveats lie and which features are optional (for minimalistic implementations)

File: source/plugin/economy/basics.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+- **Unique Accounts**
+
+ - Unique accounts are tied to a UUID, usually a player. To get the UUID of an unique account, 
+   use ``getUUID()``.
+
+Transactions
+~~~~~~~~~~~~
+
+Transactions represent the changing of the balance of an account. There are currently three types of transactions:
```
Comment (oliverdunk) [inline source/plugin/economy/basics.rst]: This sounds like quite a mouthful. Perhaps "Transactions represent an account's change in balance" would be more fitting?

File: source/plugin/economy/basics.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+~~~~~~~~~~~~
+
+Transactions represent the changing of the balance of an account. There are currently three types of transactions: 
+
+- Deposit: Occurs when an account has funds added to it
+- Withdraw: Occurs when an account has funds removed from it
+- Transfer: Occurs when an account exchanges funds with another account
+
+When a transfer occurs, the ``EconomyTransactionEvent`` is fired. Using 
+the event, you can get the ``TransactionResult``. The ``TransactionResult`` stores data about the transaction that
```
Comment (oliverdunk) [inline source/plugin/economy/basics.rst]: "this event" sounds a little better.

**Comment by oliverdunk** [comment]
Added a few grammar related comments, other than that I think this is good and it represents the code style we'd expect quite effectively.

**Comment by 12AwsomeMan34** [comment]
One quirk, the 'Best Practices' page still says that the Economy API is in development.

**Comment by Flibio** [comment]
@12AwsomeMan34 Thanks for letting me know. It has been fixed.

**Comment by Tzky** [comment]
Closing in favour of #416

**Pull request closed without merging**